site/blog/2022/deploy-ghost-ec2-advanced.mdx

```diff
@@ -0,0 +1,192 @@
+ -- Install modules
+SELECT * FROM iasql_install ('aws_route53', 'aws_ec2_metadata');
+
+-- Grab public IP for previously created instance
+SELECT public_ip_address FROM instance_metadata WHERE id = (
+    SELECT id FROM instance WHERE tags ->> 'name' = 'ghost-blog' LIMIT 1
+  );
+
+-- Create recordset
+SELECT * FROM iasql_begin();
+INSERT INTO hosted_zone (domain_name) VALUES ('test-ghost.com.');
+INSERT INTO resource_record_set (name, parent_hosted_zone_id, record_type, record, ttl) 
+SELECT 'blog.test-ghost.com.', id , 'A', '<instance_public_ip>', 300
+FROM hosted_zone where domain_name='test-ghost.com.';
+SELECT * FROM iasql_commit();
+```
+
+After some period of time, and assuming that the domain nameservers are pointing to <a href="https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/migrate-dns-domain-in-use.html#migrate-dns-change-name-servers-with-provider">AWS nameservers</a>,
+the `blog.test-ghost.com` subdomain should be resolved to the EC2 instance IP.
+
+Finally, Ghost needs to be configured to serve this new domain. This can be achieved by connecting by SSH to the instance:
+
+```console
+ssh ubuntu@<instance_public_ip>
+sudo su ghostuser
+cd /var/www/ghost
+ghost config url http://blog.test-ghost.com
+ghost setup nginx
+ghost restart
+```
+
+## Serve traffic over HTTPS
+
+The current deployment is served as HTTP, something not really suitable for production, where all traffic needs to be served via HTTPS. In order to achieve that, we can request a certificate for the
+domain using <a href="https://aws.amazon.com/certificate-manager">AWS ACM</a> functionality. The certificate will be validated automatically using the <a href="https://docs.aws.amazon.com/acm/latest/userguide/dns-validation.html">DNS method</a>:
+
+```sql TheButton[Request certificate and validate via DNS]="Request certificate and validate via DNS"
+ -- Install modules
+SELECT * FROM iasql_install ('aws_acm');
+
+-- request certificate
+SELECT * FROM certificate_request('test-ghost.com', 'DNS', 'us-east-2', '');
+```
+
+The DNS method is performed automagically in IaSQL if the domain is managed via Route 53 service. As an alternative, manual validation can be issued via this method, or e-mail.
+In the e-mail case, a validation link will be sent to the associated e-mail for the domain. The certificate will be activated by manually clicking on the activation link:
+
+```sql TheButton[Request certificate and validate via email]="Request certificate and validate via email"
+ -- Install modules
+SELECT * FROM iasql_install ('aws_acm');
+
+-- request certificate
+SELECT * FROM certificate_request('test-ghost.com', 'EMAIL', 'us-east-2', '');
+```
+
+Once the certificate has been properly issued, it needs to be associated with the current EC2 instance. This is achieved by creating a
+<a href="https://aws.amazon.com/elasticloadbalancing/">Load Balancer</a> - application, internet facing - and properly associating the certificate and domain:
```

Comment:
It could be fed into the `nginx` server without involving the load balancer, to be fair. Perhaps we should state that the easiest way to use the certificate is with the AWS-provided load balancer?